src/GovUk.Education.ExploreEducationStatistics.Content.Services/PublicationService.cs

```diff
@@ -0,0 +1,70 @@
+﻿#nullable enable
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using GovUk.Education.ExploreEducationStatistics.Common.Cache;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Utils;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.ViewModels;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace GovUk.Education.ExploreEducationStatistics.Content.Services;
+
+public class PublicationService : IPublicationService
+{
+    private readonly IPersistenceHelper<ContentDbContext> _contentPersistenceHelper;
+    private readonly IMapper _mapper;
+
+    public PublicationService(
+        IPersistenceHelper<ContentDbContext> contentPersistenceHelper,
+        IMapper mapper)
+    {
+        _contentPersistenceHelper = contentPersistenceHelper;
+        _mapper = mapper;
+    }
+
+    [BlobCache(typeof(PublicationCacheKey))]
+    public async Task<Either<ActionResult, PublicationViewModel>> Get(string publicationSlug)
+    {
+        return await _contentPersistenceHelper
+            .CheckEntityExists<Publication>(query => query
+                .Include(p => p.Releases)
+                .Include(p => p.Contact)
+                .Include(p => p.LegacyReleases)
+                .Include(p => p.Topic)
+                .ThenInclude(topic => topic.Theme)
+                .Where(p => p.Slug == publicationSlug))
+            .OnSuccess(publication =>
+            {
+                var publicationViewModel = _mapper.Map<PublicationViewModel>(publication);
+                // NOTE: BlobCache won't cache result if return a Either.IsLeft - if there is no latestRelease
+                return GetLatestRelease(publication)
```

Comment:
`var publicationViewModel = _mapper.Map<PublicationViewModel>(publication);` could move inside the `OnSuccess` below as it's not relevant if there's no latest Release.

Alternatively `OnSuccess(publication` above could change to `OnSuccessCombineWith` and then get the latest release and then you would reduce the level of nesting here down to one block where you have the publication and latest release, and then build the view model.

Reply:
Made the change. Had to create a new overload of `OnSuccessCombineWith` that doesn't return a Task to get this to work.